includes/express-checkout/class-wc-payments-express-checkout-button-helper.php

```diff
@@ -232,4 +241,153 @@ private function get_quantity() {
+	 *
+	 * @return boolean
+	 */
+	public function is_cart() {
+		return is_cart() || has_block( 'woocommerce/cart' );
+	}
+
+	/**
+	 * Checks if this is the checkout page or content contains a cart block.
+	 *
+	 * @return boolean
+	 */
+	public function is_checkout() {
+		return is_checkout() || has_block( 'woocommerce/checkout' );
+	}
+
+	/**
+	 * Checks if button is available at a given location.
+	 *
+	 * @param string $location Location.
+	 * @param string $option_name Option name.
+	 * @return boolean
+	 */
+	public function is_available_at( $location, $option_name ) {
+		$available_locations = $this->gateway->get_option( $option_name );
+		if ( $available_locations && is_array( $available_locations ) ) {
+			return in_array( $location, $available_locations, true );
+		}
+
+		return false;
+	}
+
+	/**
+	 * The settings for the `button` attribute - they depend on the "grouped settings" flag value.
+	 *
+	 * @return array
+	 */
+	public function get_button_settings() {
+		$button_type = $this->gateway->get_option( 'payment_request_button_type' );
+		return [
+			'type'         => $button_type,
+			'theme'        => $this->gateway->get_option( 'payment_request_button_theme' ),
+			'height'       => $this->get_button_height(),
+			// Default format is en_US.
+			'locale'       => apply_filters( 'wcpay_payment_request_button_locale', substr( get_locale(), 0, 2 ) ),
+			'branded_type' => 'default' === $button_type ? 'short' : 'long',
+			'size'         => $this->gateway->get_option( 'payment_request_button_size' ),
+			'context'      => $this->get_button_context(),
+		];
+	}
+
+	/**
+	 * Gets the context for where the button is being displayed.
+	 *
+	 * @return string
+	 */
+	public function get_button_context() {
+		if ( $this->is_product() ) {
+			return 'product';
+		}
+
+		if ( $this->is_cart() ) {
+			return 'cart';
+		}
+
+		if ( $this->is_pay_for_order_page() ) {
+			return 'pay_for_order';
+		}
+
+		if ( $this->is_checkout() ) {
+			return 'checkout';
+		}
+
+		return '';
+	}
+
+	/**
+	 * Gets the button height.
+	 *
+	 * @return string
+	 */
+	public function get_button_height() {
+		$height = $this->gateway->get_option( 'payment_request_button_size' );
+		if ( 'medium' === $height ) {
+			return '48';
+		}
+
+		if ( 'large' === $height ) {
+			return '56';
+		}
+
+		// for the "default"/"small" and "catch-all" scenarios.
+		return '40';
+	}
+
+	/**
+	 * Get product from product page or product_page shortcode.
+	 *
+	 * @return WC_Product|false|null Product object.
+	 */
+	public function get_product() {
+		global $post;
+
+		if ( is_product() ) {
+			return wc_get_product( $post->ID );
+		} elseif ( wc_post_content_has_shortcode( 'product_page' ) ) {
+			// Get id from product_page shortcode.
+			preg_match( '/\[product_page id="(?<id>\d+)"\]/', $post->post_content, $shortcode_match );
+			if ( isset( $shortcode_match['id'] ) ) {
+				return wc_get_product( $shortcode_match['id'] );
+			}
+		}
+
+		return null;
```

Comment:
WooPay button handler returns `false` and thus misaligns with the rest. I think returning `null` is OK here but might be worth checking if possible what's the reason behind WooPay button handler returning `false` here.

Reply:
Nice catch! I should've noted on this one. So it looks like the PRB `get_product()` from which WooPay borrowed it's `get_product()` originally returned `false`. Then, it was[ later updated to return `null`](https://github.com/Automattic/woocommerce-payments/pull/6377/files#diff-01da8e7c10c31988f0205e9327a25bcaaf64e6485c6d391912b68dc66d808353) to handle an edge case that wasn't reproducible.

>I think returning null is OK here

I agree. `null` will fail all checks in `is_product_supported()` [in the WooPay button handler](https://github.com/Automattic/woocommerce-payments/blob/bce850c7d4eca58833e2a1f34426f31c3c838277/includes/class-wc-payments-woopay-button-handler.php#L340) ensuring `$is_supported` is false if this were to occur.